### PR TITLE
Fix dislocated parent tasks in FactoryBot - Part 2

### PR DESCRIPTION
**Pull request by yoomlam**
Connects #13130
Part 1 #13604

### Description
Second, collect and group all AMA tasks together under a common parent factory, in preparation for next part.

### Acceptance Criteria
- [ ] All tests pass

### CODE REVIEW

File: spec/factories/task.rb [hunk 1]
```diff
@@ -299,260 +236,283 @@
...
+    factory :higher_level_review_task, class: DecisionReviewTask do
+      type { DecisionReviewTask.name }
+      appeal { create(:higher_level_review, benefit_type: "education") }
+      assigned_by { nil }
     end
 
-    factory :change_hearing_disposition_task, class: ChangeHearingDispositionTask do
-      type { ChangeHearingDispositionTask.name }
-      assigned_to { HearingAdmin.singleton }
+    factory :ama_task_base_factory do
```
Comment (yoomlam) [inline spec/factories/task.rb]: New parent factory to collect all AMA tasks.

File: spec/factories/task.rb [hunk 2]
```diff
@@ -299,260 +236,283 @@
...
 
-    factory :assign_hearing_disposition_task, class: AssignHearingDispositionTask do
-      type { AssignHearingDispositionTask.name }
-      assigned_to { Bva.singleton }
-      appeal
-      parent { create(:hearing_task) }
+    factory :higher_level_review_task, class: DecisionReviewTask do
+      type { DecisionReviewTask.name }
+      appeal { create(:higher_level_review, benefit_type: "education") }
+      assigned_by { nil }
```
Comment (yoomlam) [inline spec/factories/task.rb]: Deal with these later (in a subsequent stacked PR).

File: spec/factories/task.rb [hunk 3]
```diff
@@ -299,260 +236,283 @@
...
+    factory :ama_task_base_factory do
       appeal
-    end
 
-    factory :ama_judge_decision_review_task, class: JudgeDecisionReviewTask do
-      type { JudgeDecisionReviewTask.name }
-      appeal
-    end
+      factory :ama_colocated_task, traits: [ColocatedTask.actions_assigned_to_colocated.sample.to_sym],
+                                   parent: :colocated_task do
```
Comment (hschallhorn) [inline spec/factories/task.rb]: Didn't know we could call a different parent. 👌

File: spec/factories/task.rb [hunk 4]
```diff
@@ -299,260 +236,283 @@
...
+        end
+      end
 
-    factory :translation_task, class: TranslationTask do
-      type { TranslationTask.name }
-      appeal
-    end
+      factory :ama_task, class: Task do
+        type { Task.name }
+      end
```
Comment (hschallhorn) [inline spec/factories/task.rb]: Could we use this task as the parent factory instead as it already serves as a base "generic task on an ama appeal"?

Comment (yoomlam) [reply spec/factories/task.rb]: Sure! My only concern is `class: Task`, i.e., if child factories will override it properly. I created a parent `ama_task_base_factory` in Part 3 so I'll execute your suggestion there.

**Pull request merged**